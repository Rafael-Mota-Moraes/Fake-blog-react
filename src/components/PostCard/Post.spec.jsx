const { render, screen } = require("@testing-library/react");
import { PostCard } from ".";
import mock from "./mock";

describe("<PostCard />", () => {
  it("should render PostCard correctly", () => {
    const { debug } = render(<PostCard {...mock} />);
    expect(screen.getByRole("img", { name: /TÃ­tulo 1/i })).toHaveAttribute(
      "src",
      mock.cover
    );

    expect(screen.getByText(mock.title)).toBeInTheDocument();
  });
  it("should match snapshot", () => {
    const { container } = render(<PostCard {...mock} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
