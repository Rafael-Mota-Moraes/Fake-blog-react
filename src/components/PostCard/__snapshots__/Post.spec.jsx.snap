// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Título 1"
    class="post-image"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h1
      class="post-title"
    >
      Título 1
    </h1>
    <p>
      Body 1
      ...
    </p>
  </div>
</div>
`;
